src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -657,15 +657,6 @@ br_status seq_rewriter::mk_seq_concat(expr* a, expr* b, expr_ref& result) {
         result = a;
         return BR_DONE;
     }
-    // TBD concatenation is right-associative
-    if (isc2 && m_util.str.is_concat(a, c, d) && m_util.str.is_string(d, s1)) {
-        result = m_util.str.mk_concat(c, m_util.str.mk_string(s1 + s2));
-        return BR_DONE;
-    }
-    if (isc1 && m_util.str.is_concat(b, c, d) && m_util.str.is_string(c, s2)) {
```

Comment:
this one is still a reachable case: "a" + ("b" + x) -> "ab" + x

Reply:
Ah you are right, only the `is_concat(a, c, d)` case is unreachable. Fixed